libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -48,55 +61,86 @@ class MapboxRouteLineView(var options: MapboxRouteLineOptions) {
      * @param routeDrawData a [Expected<RouteLineError, RouteSetValue>]
      */
     fun renderRouteDrawData(style: Style, routeDrawData: Expected<RouteLineError, RouteSetValue>) {
-        initializeLayers(style, options)
+        routeDrawData.value?.let { value ->
+            jobControl.scope.launch {
+                mutex.withLock {
+                    initializeLayers(style, options)
 
-        routeDrawData.onValue {
-
-            updateLineGradient(
-                style,
-                RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID,
-                it.trafficLineExpression
-            )
-            updateLineGradient(
-                style,
-                RouteLayerConstants.PRIMARY_ROUTE_LAYER_ID,
-                it.routeLineExpression
-            )
-            updateLineGradient(
-                style,
-                RouteLayerConstants.PRIMARY_ROUTE_CASING_LAYER_ID,
-                it.casingLineExpression
-            )
-            updateSource(
-                style,
-                RouteConstants.PRIMARY_ROUTE_SOURCE_ID,
-                it.primaryRouteSource
-            )
-            updateSource(
-                style,
-                RouteConstants.ALTERNATIVE_ROUTE1_SOURCE_ID,
-                it.alternativeRoute1Source
-            )
-            updateSource(
-                style,
-                RouteConstants.ALTERNATIVE_ROUTE2_SOURCE_ID,
-                it.alternativeRoute2Source
-            )
-            updateLineGradient(
-                style,
-                RouteLayerConstants.ALTERNATIVE_ROUTE1_TRAFFIC_LAYER_ID,
-                it.altRoute1TrafficExpression
-            )
-            updateLineGradient(
-                style,
-                RouteLayerConstants.ALTERNATIVE_ROUTE2_TRAFFIC_LAYER_ID,
-                it.altRoute2TrafficExpression
-            )
-            updateSource(
-                style,
-                RouteConstants.WAYPOINT_SOURCE_ID,
-                it.waypointsSource
-            )
+                    updateLineGradient(
+                        style,
+                        RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID,
+                        Expression.color(Color.TRANSPARENT)
+                    )
```

Comment:
This also has to be done for alternatives.

Reply:
d'oh! I should have caught that.